internal/providers/github/github.go

```diff
@@ -115,3 +126,78 @@ func ParseV1Config(rawCfg json.RawMessage) (*minderv1.GitHubProviderConfig, erro
 
 	return w.GitHub, nil
 }
+
+func (c *RestClient) waitIfRateLimited(ctx context.Context) error {
+	blockedChan, ok := c.tokenStatStore.GetBlockingChan(c.token, db.ProviderTypeGithub)
+	if !ok {
+		return nil
+	}
+
+	select {
+	case <-blockedChan:
+		return nil
+	case <-ctx.Done():
+		return ctx.Err()
+	case <-time.After(MaxRateLimitWait):
+		return fmt.Errorf("rate limit wait exceeded for %s", c.GetOwner())
```

Comment:
Do you think it would be a good idea to create a timer manually and cancel it the wait finishes sooner than the timer fires to deallocate it?
```go
timer := time.NewTimer(MaxRateLimitWait)
    defer func() {
        if !timer.Stop() {
            <-timer.C
        }
    }()
```

Reply:
Yeah, we should stop the timer with defer for better and faster gc. How about
```go
func (c *RestClient) waitIfRateLimited(ctx context.Context) error {
	blockedChan, ok := c.tokenStatStore.GetBlockingChan(c.token, db.ProviderTypeGithub)
	if !ok {
		return nil
	}

	timer := time.NewTimer(MaxRateLimitWait)
	defer timer.Stop()

	select {
	case <-blockedChan:
		return nil
	case <-ctx.Done():
		return ctx.Err()
	case <-timer.C:
		return fmt.Errorf("rate limit wait exceeded for %s", c.GetOwner())
	}
}
```

Reply:
This looks good!